### PR TITLE
FIXES #276 Admin tax rounding not the same as catalog

### LINKED ISSUE #276: Admin Tax Rounding not the same as Catalog
Scenario
Tax is set at 20%. 
Product price including tax is 6.99 which when entered into Products Price (Gross) on the product entry page is calculated as 5.825 in the Products Price (Net) entry field. 
Preview page changes this to 2 decimal places making it 5.83. 
Tax added to 5.83 is 7.00.
Tax added to 5.825 is 6.99

The product list in admin shows the product as 7.00 as do any order pages and invoices. 
Store side shows the correct value of 6.99

### PR DESCRIPTION
**Pull request by scottcwilson**
Using the example from the issue, bot the catalog and admin
show the price at $6.99.

### CODE REVIEW

File: includes/functions/functions_taxes.php [hunk 1]
```diff
@@ -166,9 +166,9 @@ function zen_add_tax($price, $tax) {
 
     if (IS_ADMIN_FLAG === true) {
       if (DISPLAY_PRICE_WITH_TAX_ADMIN == 'true') {
-        return zen_round($price, $currencies->currencies[DEFAULT_CURRENCY]['decimal_places']) + zen_calculate_tax($price, $tax);
+        return $price + zen_calculate_tax($price, $tax);
       } else {
-        return zen_round($price, $currencies->currencies[DEFAULT_CURRENCY]['decimal_places']);
+        return $price;
```
Comment (drbyte) [inline includes/functions/functions_taxes.php]: I'm assuming these changes are here to mirror the `else` condition on the following lines.

File: includes/classes/order.php [hunk 2]
```diff
@@ -497,7 +497,7 @@ protected function cart(shoppingCart $basket, $customer_id = null, $sendto_addre
                                       'tax_groups'=>$taxRates,
                                       'tax_description' => zen_get_tax_description($products[$i]['tax_class_id'], $taxCountryId, $taxZoneId),
                                       'price' => $products[$i]['price'],
-                                      'final_price' => zen_round($products[$i]['price'] + $basket->attributes_price($products[$i]['id']), $decimals),
+                                      'final_price' => $products[$i]['price'] + $basket->attributes_price($products[$i]['id']),
```
Comment (drbyte) [inline includes/classes/order.php]: why was this change made?

File: includes/classes/currencies.php [hunk 3]
```diff
@@ -91,7 +91,7 @@ function value($number, $calculate_currency_value = true, $currency_type = '', $
       }
       $currency_value = zen_round($number * $rate, $this->currencies[$currency_type]['decimal_places']);
     } else {
-      $currency_value = zen_round($number, $this->currencies[$currency_type]['decimal_places']);
+      $currency_value = $number;
```
Comment (drbyte) [inline includes/classes/currencies.php]: Why this change?

File: includes/functions/functions_taxes.php (hunk 1 above)
Comment (scottcwilson) [inline includes/functions/functions_taxes.php]: Correct

File: includes/classes/order.php (hunk 2 above)
Comment (scottcwilson) [inline includes/classes/order.php]: Rounding here is premature. You still have to multiply by quantity.

File: includes/classes/currencies.php (hunk 3 above)
Comment (scottcwilson) [inline includes/classes/currencies.php]: Sorry, that was a failed experiment I should have removed.  Fixed.

**Comment by drbyte** [comment]
I'd like to hear @zcwilt feedback after running the webtest suite against this.

**Comment by zcwilt** [comment]
:+1:
I'm still having problems running webtests due to an incompatibility between the latest firefox version and the selenium/marionette driver. However unit tests do contain some pricing checks (mainly discount coupon price/tax stuff) 
given I cant see any probs logic wise, lets pass this pr. I'll look to move some other pricing checks from web tests to unit tests shortly.

**Comment by scottcwilson** [comment]
Can we merge?

**Comment by drbyte** [comment]
👍
